lib/core-integration/src/Test/Integration/Scenario/API/Transactions.hs

```diff
@@ -1025,26 +1043,338 @@ spec = do
+        let insIn = getFromResponseList 1 inputs r
+        length (outsIn :: [AddressAmount t]) `shouldBe` 1000
+        length (insIn :: [ApiTxInput t]) `shouldBe` 1
+
+        verify r
+            [ expectResponseCode @IO HTTP.status200
+            , expectListSizeEqual 2
+            , expectListItemFieldEqual 0 direction Outgoing
+            , expectListItemFieldBetween 0 amount (feeMin + 1, feeMax + 1)
+            , expectListItemFieldEqual 0 status InLedger
+            , expectListItemFieldEqual 1 direction Incoming
+            , expectListItemFieldEqual 1 amount 1_000_000_000_000
+            , expectListItemFieldEqual 1 status InLedger
+            ]
+
+    -- This scenario covers the following matrix of cases. Cases where generated
+    -- using one of parwise test cases generation tools available online.
+    -- +----+----------+----------+------------+--------------+
+    --     |  start   |   end    |   order    |    result    |
+    -- +----+----------+----------+------------+--------------+
+    --   1 | edge     | edge     | ascending  | 2 ascending  |
+    --   2 | edge     | edge + 1 | descending | 2 descending |
+    --   3 | edge     | edge - 1 | empty      | 1st one      |
+    --   4 | edge     | empty    | empty      | 2 descending |
+    --   5 | edge + 1 | edge + 1 | empty      | 2nd one      |
+    --   6 | edge + 1 | edge - 1 | empty      | none         |
+    --   7 | edge + 1 | empty    | ascending  | 2nd one      |
+    --   8 | edge + 1 | edge     | descending | 2nd one      |
+    --   9 | edge - 1 | edge - 1 | ascending  | 1st one      |
+    --  10 | edge - 1 | empty    | descending | 2 descending |
+    --  11 | edge - 1 | edge     | empty      | 2 descending |
+    --  12 | edge - 1 | edge + 1 | empty      | 2 descending |
+    --  13 | empty    | empty    | empty      | 2 descending |
+    --  14 | empty    | edge     | empty      | 2 descending |
+    --  15 | empty    | edge + 1 | ascending  | 2 ascending  |
+    --  16 | empty    | edge - 1 | descending | 1st one      |
+    -- +----+----------+----------+------------+--------------+
+    it "TRANS_LIST_02,03 - Can limit/order results with start, end and order" $ \ctx -> do
+        (_, w, [(t1, a1), (t2, a2)]) <- fixtureWalletManyTxs ctx [10,20]
+
+        let toParam = toQueryParam . utcIso8601ToText
+        let matrix =
+                [
+                -- 1
+                  ( "?start="
+                    <> (toParam t1)
+                    <> "&end="
+                    <> (toParam t2)
+                    <> "&order=ascending"
+                  ,  [ expectListSizeEqual 2
+                     , expectListItemFieldEqual 0 amount a1
+                     , expectListItemFieldEqual 1 amount a2
+                     ]
+                  )
+                -- 2
+                , ( "?start="
+                    <> (toParam t1)
+                    <> "&end="
+                    <> (toParam $ addUTCTime 1 t2)
```

Comment:
If you're trying to get the next `UTCTime` value after `t2` here, then you could use the following construction:
```hs
<> (toParam $ addUTCTime (succ 0) t2)
```
In this case, the `Enum` instance guarantees not to skip any intermediate values.

On the other hand, specifying `1` here will actually get you an increment of `10^12` picoseconds, because while `NominalDiffTime` does have a `Num` instance, it multiplies the value internally using the functionality provided by `Data.Fixed`:

http://hackage.haskell.org/package/base-4.12.0.0/docs/src/Data.Fixed.html#line-100

Reply:
> If you're trying to get the next UTCTime value after t2 here, then you could use the following construction:

Actually I was going for getting the next UTCTime incremented/decremented by exactly 1 second (i.e. `10^12` picoseconds), since I think it'd be the most common usecase. 

We already have `TIME_RANGE_LIMIT` tests which tests small ranges (in `picoseconds`) implemented by you, so I we cover that. I think that both are somewhat complementary.

Reply:
Fair enough! My question was mostly to figure out your intentions here. I think the existing tests provide coverage that's good enough for our needs. :)